MSAL/test/integration/native_auth/end_to_end/sign_out/MSALNativeAuthSignOutEndToEndTests.swift

```diff
@@ -25,65 +25,20 @@
-        guard let sut = initialisePublicClientApplication() else {
-            XCTFail("Missing information")
-            return
-        }
-
-        let signInExpectation = expectation(description: "signing in")
-        let verifyCodeExpectation = expectation(description: "verifying code")
-        let signInDelegateSpy = SignInStartDelegateSpy(expectation: signInExpectation)
-        let signInVerifyCodeDelegateSpy = SignInVerifyCodeDelegateSpy(expectation: verifyCodeExpectation)
-
-        let username = ProcessInfo.processInfo.environment["existingOTPUserEmail"] ?? "<existingOTPUserEmail not set>"
-        let otp = "<otp not set>"
-
-        sut.signIn(username: username, correlationId: correlationId, delegate: signInDelegateSpy)
-
-        await fulfillment(of: [signInExpectation])
-
-        XCTAssertTrue(signInDelegateSpy.onSignInCodeRequiredCalled)
-        XCTAssertNotNil(signInDelegateSpy.newStateCodeRequired)
-        XCTAssertNotNil(signInDelegateSpy.sentTo)
-
-        // Now submit the code..
-
-        signInDelegateSpy.newStateCodeRequired?.submitCode(code: otp, delegate: signInVerifyCodeDelegateSpy)
-
-        await fulfillment(of: [verifyCodeExpectation])
-
-        XCTAssertTrue(signInVerifyCodeDelegateSpy.onSignInCompletedCalled)
-        XCTAssertNotNil(signInVerifyCodeDelegateSpy.result)
-        XCTAssertNotNil(signInVerifyCodeDelegateSpy.result?.idToken)
-        XCTAssertEqual(signInVerifyCodeDelegateSpy.result?.account.username, username)
-
-        // Sign out
-
-        var userAccountResult = sut.getNativeAuthUserAccount()
-        userAccountResult?.signOut()
-
-        userAccountResult = sut.getNativeAuthUserAccount()
-        XCTAssertNil(userAccountResult)
-    }
-
+final class MSALNativeAuthSignOutEndToEndTests: MSALNativeAuthEndToEndPasswordTestCase {
     // Hero Scenario 2.4.1. Sign out – Local sign out from app on device (no SSO)
     func test_signOutAfterSignInPasswordSuccess() async throws {
-        throw XCTSkip("Skipping this test because native auth KeyVault is missing")
-        guard let sut = initialisePublicClientApplication() else {
+        guard let sut = initialisePublicClientApplication(),
+              let username = retrieveUsernameForSignInUsernameAndPassword(),
+              let password = await retrievePasswordForSignInUsername()
+        else {
             XCTFail("Missing information")
             return
         }
         
         let signInExpectation = expectation(description: "signing in")
         let signInDelegateSpy = SignInPasswordStartDelegateSpy(expectation: signInExpectation)
 
-        let username = ProcessInfo.processInfo.environment["existingPasswordUserEmail"] ?? "<existingPasswordUserEmail not set>"
-        let password = ProcessInfo.processInfo.environment["existingUserPassword"] ?? "<existingUserPassword not set>"
-
         sut.signIn(username: username, password: password, correlationId: correlationId, delegate: signInDelegateSpy)
```

Comment:
It doesn't appear in the diff, but before the second line (`userAccountResult?.signOut()`), could you add an assert to make sure the account is not nil?

```
        // Sign out

        var userAccountResult = sut.getNativeAuthUserAccount()
        // add here XCTAssertNotNil(userAccountResult)
        userAccountResult?.signOut()

        userAccountResult = sut.getNativeAuthUserAccount()
        XCTAssertNil(userAccountResult)
```

Reply:
Added.